src/PatchOrchestrationApplication/CoordinatorService/src/RepairManagerHelper.cs

```diff
@@ -274,6 +279,214 @@ internal async Task PrintRepairTasks(CancellationToken cancellationToken)
+        /// <parameterName>Name of the parameter to search.</parameterName>
+        /// <return>the value of parameter</return>
+        public string GetParamValueFromSection(XmlDocument doc, string sectionName, string parameterName)
+        {
+            XmlNode sectionNode = doc.DocumentElement?.SelectSingleNode("//*[local-name()='Section' and @Name='" + sectionName + "']");
+            XmlNode parameterNode = sectionNode?.SelectSingleNode("//*[local-name()='Parameter' and @Name='" + parameterName + "']");
+            XmlAttribute attr = parameterNode?.Attributes?["Value"];
+            return attr?.Value;
+        }
+
+        /// <summary>
+        /// Used to clear the events on Coordinator Service for nodes which are deleted from cluster.
+        /// </summary>
+        public async Task ClearOrphanEvents(CancellationToken cancellationToken)
+        {
+            try
+            {
+                ServiceHealth health = await this.fabricClient.HealthManager.GetServiceHealthAsync(this.context.ServiceName);
+                List<HealthEvent> healthEventsToCheck = new List<HealthEvent>();
+                foreach (var e in health.HealthEvents)
+                {
+                    if (e.HealthInformation.Property.Contains(WUOperationStatusUpdate))
+                    {
+                        healthEventsToCheck.Add(e);
+                    }
+                }
+                cancellationToken.ThrowIfCancellationRequested();
+
+                NodeList nodeList = await this.fabricClient.QueryManager.GetNodeListAsync(null, null, this.DefaultTimeoutForOperation, cancellationToken);
+                List<string> orphanProperties = new List<string>();
+                Dictionary<string, bool> propertyDict = new Dictionary<string, bool>();
+                if (healthEventsToCheck.Count == nodeList.Count)
+                {
+                    return;
+                }
+                else
+                {
+                    foreach (var node in nodeList)
+                    {
+                        propertyDict.Add(WUOperationStatusUpdate + "-" + node.NodeName, true);
+                    }
+                    foreach (var e in healthEventsToCheck)
+                    {
+                        if (!propertyDict.ContainsKey(e.HealthInformation.Property))
+                        {
+                            orphanProperties.Add(e.HealthInformation.Property);
+                        }
+                    }
+
+                    foreach (var property in orphanProperties)
+                    {
+                        ServiceEventSource.Current.VerboseMessage("Property {0}'s event is removed from CoordinatorService", property);
+
+                        // I think we would need to change the expiry time to ~0
+                        string description = "This health event will be expired in 1 minute as node corresponding to this event is deleted.";
```

Comment:
>This health event will be expired in 1 minute as node corresponding to this event is deleted. [](start = 46, length = 93)

We should just update it with message - "This node is no longer part of the cluster."